### DISCUSSION TITLE
module can't initialize EG25-E

### DISCUSSION THREAD

**Discussion by hellkaim** [IchthysMaranatha/asterisk-chan-quectel#33] [General]
Hi. I compiled and installed the module under Asterisk 13.38.1
After initialization it produces the error as follwos:

_pbx*CLI> module load chan_quectel.so
Loaded chan_quectel.so
[May 24 17:47:33] NOTICE[2333]: chan_quectel.c:1819 reload_config: [quectel0] Loaded device
  == Registered channel type 'Quectel' (Channel Driver for Mobile Telephony)
    -- [quectel0] Trying to connect on /dev/ttyUSB3...
    -- [quectel0] Quectel has connected, initializing...
  == Registered application 'QuectelStatus'
  == Registered application 'QuectelSendSMS'
  == Registered application 'QuectelSendUSSD'
  == Manager registered action QuectelShowDevices
  == Manager registered action QuectelSendUSSD
  == Manager registered action QuectelSendSMS
  == Manager registered action QuectelSetCCWA
  == Manager registered action QuectelReset
  == Manager registered action QuectelRestart
  == Manager registered action QuectelStop
  == Manager registered action QuectelStart
  == Manager registered action QuectelRemove
  == Manager registered action QuectelReload
 Loaded chan_quectel.so => (Channel Driver for Mobile Telephony)
[May 24 17:47:38] ERROR[2336]: chan_quectel.c:670 do_monitor_phone: [quectel0] timedout while waiting 'OK' in response to 'AT'
    -- [quectel0] Error initializing Quectel
    -- [quectel0] Quectel has disconnected
    -- [quectel0] Trying to connect on /dev/ttyUSB3...
    -- [quectel0] Quectel has connected, initializing...
[May 24 17:47:53] ERROR[2340]: chan_quectel.c:670 do_monitor_phone: [quectel0] timedout while waiting 'OK' in response to 'ATZ'
    -- [quectel0] Error initializing Quectel
    -- [quectel0] Quectel has disconnected_

I can connect to the ttyUSB device using screen ans it seems to work just fine:

_@es-pbx:~$ sudo screen /dev/ttyUSB3
AT
OK
ATZ
OK
AT+CGMM
EG25

OK
AT+GMI
Quectel

OK_

I know that it may well be the power issue but why then it is ok using screen and not ok using chan driver?

**Comment by IchthysMaranatha** [comment]
ttyusb3 is not the port which presents urc, if you call your own no, are you getting ring message on ttyusb3?
Please read and follow these first, I won't be able to help otherwise.
https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/4
https://github.com/IchthysMaranatha/asterisk-chan-quectel/discussions/13

**Reply by hellkaim** [reply]
Hi, sorry for delay!

I checked the modem with the `screen` command

The result is:
```
AT
OK

AT+CPIN?
+CPIN: READY

OK

AT+CGREG?
+CGREG: 0,1

OK

AT+COPS?
+COPS: 0,0,"Orange Orange",2

OK

AT+CSQ?
+CSQ: 10,99

OK

AT+CNUM?
+CNUM: ,"+12123456789",145 (the correct number is shown)

OK

```

So ` ATD+121234567; ` works fine, modem establishes the connection. The incoming call seems to work (I can't see anything on the terminal, but my phone reports that it reached the SIM terminal and the light on the modem indicates it received the call) despite nothing in the terminal (should it be there?).

Maybe I should have some other firmware for my device? Currently I have:

```
AT+CGMR
EG25GGBR07A08M2G

AT+CFUN
+CFUN: 1
```

**Reply by hellkaim** [reply]
Also

```
pbx*CLI> quectel show devices
ID           Group State      RSSI Mode Submode Provider Name  Model      Firmware          IMEI             IMSI             Number        
quectel0     0     GSM not re 11   0    0       Orange SP Oran EG25       EG25GGBR07A08M2G  816737280665274  236141875803652  +34646974255
```
And it never shows Ready despite the `screen` result above (registered)

**Comment by IchthysMaranatha** [comment]
If you are not receiving ring urc, it won't work for incoming calls with chan_quectel. Port may be wrong or some one time initialization commands may be needed https://github.com/IchthysMaranatha/asterisk-chan-quectel/pull/28

creg is 0,1 so it should show registered. Are you using a modem terminal simultaneously while using chan_quectel? That will prevent proper working. Please use the latest code with imei detection and test and close all manual terminals when chan_quectel is loaded.

**Reply by hellkaim** [reply]
_// Are you using a modem terminal simultaneously while using chan_quectel?_
Sure I am not, not my first day with dongle modules for Asterisk (I have about 7 2G modems across several PBXs).

How can I add mentioned commands #28 ?

The manual result is:

```
pbx*CLI> quectel cmd quectel0 AT+QURCCFG=\"urcport\",\"usbat\"\r"
[quectel0] 'AT+QURCCFG="urcport","usbat"r' AT command queued
 [quectel0] Got Response for user's command:'OK'
[May 31 15:23:57] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"all\",1,0\r"
[quectel0] 'AT+QINDCFG="all",1,0r' AT command queued
 [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:06] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:06] ERROR[2416]: at_response.c:410 log_cmd_response_error: [quectel0] Received 'ERROR' for unhandled command 'USER'S'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"all\",1,0\r
[quectel0] 'AT+QINDCFG="all",1,0r' AT command queued
 [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:16] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:16] ERROR[2416]: at_response.c:410 log_cmd_response_error: [quectel0] Received 'ERROR' for unhandled command 'USER'S'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"csq\",0,0\r
[quectel0] 'AT+QINDCFG="csq",0,0r' AT command queued
 [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:37] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:37] ERROR[2416]: at_response.c:410 log_cmd_response_error: [quectel0] Received 'ERROR' for unhandled command 'USER'S'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"ring\",1,0\r
[quectel0] 'AT+QINDCFG="ring",1,0r' AT command queued
 [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:48] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:48] ERROR[2416]: at_response.c:410 log_cmd_response_error: [quectel0] Received 'ERROR' for unhandled command 'USER'S'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"smsincoming\",1,0\r
[quectel0] 'AT+QINDCFG="smsincoming",1,0r' AT command queued
 [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:59] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'ERROR'
[May 31 15:24:59] ERROR[2416]: at_response.c:410 log_cmd_response_error: [quectel0] Received 'ERROR' for unhandled command 'USER'S'

pbx*CLI> quectel cmd quectel0 AT+QURCCFG=\"urcport\"
[quectel0] 'AT+QURCCFG="urcport"' AT command queued
 [quectel0] Got Response for user's command:'+QURCCFG: "urcport","usbat"'
[May 31 15:25:35] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'+QURCCFG: "urcport","usbat"'
 [quectel0] Got Response for user's command:'OK'
[May 31 15:25:35] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"all\"
[quectel0] 'AT+QINDCFG="all"' AT command queued
 [quectel0] Got Response for user's command:'+QINDCFG: "all",1'
[May 31 15:25:43] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'+QINDCFG: "all",1'
 [quectel0] Got Response for user's command:'OK'
[May 31 15:25:43] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"csq\"
[quectel0] 'AT+QINDCFG="csq"' AT command queued
 [quectel0] Got Response for user's command:'+QINDCFG: "csq",0'
[May 31 15:25:51] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'+QINDCFG: "csq",0'
 [quectel0] Got Response for user's command:'OK'
[May 31 15:25:51] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"ring\"
[quectel0] 'AT+QINDCFG="ring"' AT command queued
 [quectel0] Got Response for user's command:'+QINDCFG: "ring",1'
[May 31 15:26:00] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'+QINDCFG: "ring",1'
 [quectel0] Got Response for user's command:'OK'
[May 31 15:26:00] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT+QINDCFG=\"smsincoming\"
[quectel0] 'AT+QINDCFG="smsincoming"' AT command queued
 [quectel0] Got Response for user's command:'+QINDCFG: "smsincoming",1'
[May 31 15:26:08] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'+QINDCFG: "smsincoming",1'
 [quectel0] Got Response for user's command:'OK'
[May 31 15:26:08] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'

pbx*CLI> quectel cmd quectel0 AT^DSCI
[quectel0] 'AT^DSCI' AT command queued
 [quectel0] Got Response for user's command:'OK'
[May 31 15:26:17] NOTICE[2416]: at_response.c:2053 at_response: [quectel0] Got Response for user's command:'OK'
```

**Comment by IchthysMaranatha** [comment]
Sorry, saw that cgreg is 0,1, check at+creg? It may not be registered correctly.

**Reply by hellkaim** [reply]
I got it registered
```
pbx*CLI> quectel show devices from
ID           Group State      RSSI Mode Submode Provider Name  Model      Firmware          IMEI             IMSI             Number        
quectel0     0     Free       10   0    0       Orange Orange  EG25       EG25GGBR07A08M2G  865167063728274  236141875803652  +34646974255
```

**Reply by hellkaim** [reply]
To clarify: I got outgoing calls work
```
pbx*CLI> quectel show devices
ID           Group State      RSSI Mode Submode Provider Name  Model      Firmware          IMEI             IMSI             Number        
quectel0     0     Free       10   0    0       Orange Orange  EG25       EG25GGBR07A08M2G  865167063728274  236141875803652  +34646974255  
  == Setting global variable 'SIPDOMAIN' to '10.0.0.12'
    -- Executing [+34966188600@default:1] Verbose("PJSIP/gsm-00000000", "1, "User 101 dialed +34966188600."") in new stack
  "User 101 dialed +34966188600."
    -- Executing [+34966188600@default:2] Set("PJSIP/gsm-00000000", "CLID=101") in new stack
    -- Executing [+34966188600@default:3] GotoIf("PJSIP/gsm-00000000", "1?CallDongle0") in new stack
    -- Goto (default,+34966188600,6)
    -- Executing [+34966188600@default:6] Dial("PJSIP/gsm-00000000", "Quectel/quectel0/+34966188600") in new stack
    -- Called Quectel/quectel0/+34966188600
    -- Quectel/quectel0-0100000000 is making progress passing it to PJSIP/gsm-00000000
  == Spawn extension (default, +34966188600, 6) exited non-zero on 'PJSIP/gsm-00000000'
```

**Comment by IchthysMaranatha** [comment]
Is everything working now? 
Else, pass at+qindcfg="all",1,1 using modem terminal and check. Make sure inverted commas are of the plain type. With chan_quectel unloaded, connect terminal to ttyusb2 and check if you are getting ring urc

**Reply by hellkaim** [reply]
THX.

I will check the command later.
For whatever reason sometimes I have `Channel not registered` error. This well might be due to power issue as the modem is in Dialing mode and only reset helps.

Let me hack it a bit more and I will tell you the result. On the current stage I can't here any sound from or to the modem but again it well may be due to the power.

**Comment by IchthysMaranatha** [comment]
Are you checking with modem connected to a vm on a laptop/PC? That will help identifying power issues, esp if it is connected to a usb3 port. Losing regn may also be due to lack of volte config.
